networkx/algorithms/shortest_paths/ada_star.py

```diff
@@ -0,0 +1,470 @@
+
+    >>> search.compute_or_improve_path(epsilon=1.2)
+    >>> path = search.extract_path()
+    >>> print("epsilon = 1.2 path: ", path)
+    epsilon = 1.2 path:  [42, 32, 24, 12, 59, 4, 1, 27, 35, 25]
+    >>> print("epsilon = 1.2 path_weight: ", nx.path_weight(G, path, "weight"))
+    epsilon = 1.2 path_weight:  1.3335657361796027
+
+    >>> search.compute_or_improve_path(epsilon=1)
+    >>> path = search.extract_path()
+    >>> print("epsilon = 1 path: ", path)
+    epsilon = 1 path:  [42, 32, 19, 72, 49, 29, 31, 94, 35, 25]
+    >>> print("epsilon = 1 path_weight: ", nx.path_weight(G, path, "weight"))
+    epsilon = 1 path_weight:  1.29129785933092
+
+    >>> search.update_graph([[49, 97, 0]])  # set weight between 49 and 97 to 0
+    >>> search.compute_or_improve_path(epsilon=1)
+    >>> path = search.extract_path()
+    >>> print("changed epsilon = 1 path: ", path)
+    changed epsilon = 1 path:  [42, 32, 19, 72, 49, 29, 31, 94, 35, 25]
+    >>> print("changed epsilon = 1 path_weight: ", nx.path_weight(G, path, "weight"))
+    changed epsilon = 1 path_weight:  1.29129785933092
+    """
+
+    def __init__(
+        self, source, target, G, heuristic=None, weight="weight", initial_epsilon=1000
+    ):
+        """initializer for ADA* algorithm
+
+        Returns a search object, which can be used to find a path from a
+        start state to a goal state in a graph. The search object can be
+        used to iteratively improve the path over time, as well as being
+        called to update the graph with new weights. The search object, will
+        then use previous work to find a new path in the updated graph.
+
+
+        Raises
+        ------
+        NetworkXNoPath
+            If no path exists between source and target.
+
+        NodeNotFound
+            If either source or target is not in G.
+
+        """
+
+        if type(G) == nx.MultiDiGraph or type(G) == nx.MultiGraph:
+            raise nx.NetworkXNotImplemented("MultiGraph and MultiDiGraph not supported")
+
+        if source not in G or target not in G:
+            msg = f"Either source {source} or target {target} is not in G"
+            raise nx.NodeNotFound(msg)
+
+        self.source, self.target = source, target
+        if heuristic is None:
+            # The default heuristic is h=0 - same as Dijkstra's algorithm
+            def heuristic(u, v):
+                return 0
+
+        self.heursistic = heuristic
+
+        self.G = G
+        self.g, self.rhs, self.OPEN = {}, {}, {}
+
+        self.weight = _weight_function(self.G, weight)
+
+        # estimate g[n] of the cost from each state to the goal
+        self.g = {n: math.inf for n in G.nodes()}
+
+        # one-step lookahead cost, rhs[n], 0 is n is the goal, otherwise,
+        # is the minimum of the following sum:
+        # The cost between n and its neighbours plus the g value
+        # of that neighbour
+        self.rhs = {n: math.inf for n in G.nodes()}
```

Comment:
Can you rewrite this comment... and maybe choose a better variable name. What does rhs stand for?
The poor name for `g` also makes this description very difficult to follow without some more words.

Reply:
I've renamed these variables to be more understandable

g -> state_to_goal_est
rhs -> one_step_lookahead_cost